packages/core/src/package/bundle.ts

```diff
@@ -1,80 +1,89 @@
-import commonjs from '@rollup/plugin-commonjs';
-import nodeResolve from '@rollup/plugin-node-resolve';
+import path from 'path';
+
 import { cssFileFilter } from '@vanilla-extract/integration';
-import { rollup } from 'rollup';
 import type { OutputOptions } from 'rollup';
-import esbuild from 'rollup-plugin-esbuild';
+import { build as viteBuild } from 'vite';
 
 import type { EnhancedConfig } from '../config';
 import { addVanillaDebugIds, externals } from '../plugins/rollup';
 import type { Format, PackageEntryPoint } from '../types';
 import { extensionForFormat } from '../utils/files';
+import { commonViteConfig } from '../vite-config';
 
 export const createBundle = async (
   config: EnhancedConfig,
   entries: PackageEntryPoint[],
   outputOptions: OutputOptions,
 ) => {
-  const { format } = outputOptions;
+  const format = outputOptions.format as Format;
+  const extension = extensionForFormat(format);
 
-  // Vite 3 doesn't support multiple entrypoints in library mode, so we need to use rollup here directly.
-  const bundle = await rollup({
-    input: entries.map(({ entryPath }) => entryPath),
+  await viteBuild({
+    ...commonViteConfig,
     plugins: [
       addVanillaDebugIds(config.root),
-      externals(config.root),
-      nodeResolve(),
-      commonjs(),
-      esbuild({
-        jsx: 'transform',
-      }),
+      {
+        ...externals(config.root, format),
+        enforce: 'pre',
```

Comment:
Given that we own externals now, could the `pre` just be rolled into the plugin, rather than having to spread here?